src/java/org/apache/cassandra/cql3/Operator.java

```diff
@@ -290,7 +290,7 @@ public String toString()
          */
         public boolean isSatisfiedBy(AbstractType<?> type, ByteBuffer leftOperand, ByteBuffer rightOperand)
         {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException(": operation can only be computed by an indexed column with a configured analyzer");
```

Comment:
`isSatisfiedBy` is called on the replica side during execution (not necessarily on the coordinator). That means that you have an invalid prepared statement in your prepared statement cache and that the query will blow up on every request. The exception will be wrapped into other exception and will not appear as a query problem but as a request execution problem. The check must be done when the query is being prepared in `ColumnCondition.Raw::prepare`. Looking quickly at the code we might have other bugs in this areas (for LIKE for example).

Reply:
Makes sense. FWIW, LIKE should be able to compare two `ByteBuffer` given that each LIKE implementation is either prefix, suffix, contains, or matches (which seems like equals as far as I can tell). Though, the like matches implementation uses contains, so that is likely a bug.

Reply:
Yes the LIKE logic seems buggy. There is a ticket on Cassandra to fix LIKE so that it can be used for filtering. The patch has never been finished but should be easy to complete.